EIPS/eip-5089.md

```diff
@@ -0,0 +1,340 @@
+- exchangeRate: A variable used to modulate the redeemable ratio of underlying <-> Principal Token.
+- fee: An amount of underlying or Principal Token charged to the user by the Principal Token. Fees can exists on redemption or post-maturity yield. _Alberto: I'm not sure we need this in the ERC._
+
+### Methods
+
+#### underlying
+
+The address of the underlying token used by the Principal Token for accounting, depositing, and withdrawing.
+
+MUST be an ERC-20 token contract.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: underlying
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: underlyingAddress
+      type: address
+```
+
+#### maturity
+
+The unix timestamp (uint256) after which Principal Tokens can be redeemed for their underlying deposit according to `exchangeRate`.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: maturity
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: timestamp
+      type: uint256
+```
+
+#### maturityRate
+
+The exchangeRate during the block that a first user first calls `redeem` or `mature` after maturity.
+
+OPTIONAL - This method can be used to provide needed data for the calculation of interest post-maturity, but interfaces and other contracts MUST NOT expect this method to be present. _Alberto: Optionality in ERCs is better avoided. Not everyone will implement it, therefore integrators must assume it might not be there, and leads to a proliferation of `safe*` methods._
+
+MUST _NOT_ revert.
+
+```yaml
+- name: maturityRate
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: rate
+      type: uint256
+```
+
+#### exchangeRate
+
+The current exchangeRate of the underlying token <-> money market deposit post-maturity. Returns 0 before maturity.
+_Alberto: I wonder if we could make the ERC a bit more compact, and have only `exchangeRate` (renamed to `rate`). Before maturity, it returns the rate that will work at maturity, after maturity, it can return the same, or change._
+
+OPTIONAL - This method can be used to provide needed data for the calculation of interest post-maturity, but interfaces and other contracts MUST NOT expect this method to be present. _Alberto: Optionality in ERCs is better avoided. Not everyone will implement it, therefore integrators must assume it might not be there, and leads to a proliferation of `safe*` methods._
+
+MUST _NOT_ revert.
+
+```yaml
+- name: exchangeRate
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: rate
+      type: uint256
+```
+
+#### mint
+
+_Alberto: do we need a `mint` method? I reckon it might be nice for integrators to have a standard function to obtain the PTs, but so far we have been fine without one for ERC20s._
+
+Mints exactly `amount` of Principal Tokens to `to` by depositing that exact same `amount` of underlying tokens. In yield tokenization protocols, also mints `amount` of Yield Tokens in addition to Principal Tokens.
```

Comment:
This is not entirely true for yield tokenization protocols. In case there is some yield already in the pool, Yield token can have value `>0` so if you deposit `amount` of underlying you can get less PT + YT.